File: app/services/parameter_store/parameter_store.py
import boto3
from botocore.exceptions import ClientError

ssm = boto3.client("ssm", region_name="ap-southeast-2")

parameters_cache = {}

def fetch_parameter_aws(parameter_name: str) -> str | None:
    try:
        response = ssm.get_parameter(Name=parameter_name)
        return response["Parameter"]["Value"]
    except ClientError:
        return None

def fetch_parameter_local(parameter_name: str) -> str | None:
    if parameter_name in parameters_cache:
        return parameters_cache[parameter_name]

    value = fetch_parameter_aws(parameter_name)
    if value is not None:
        parameters_cache[parameter_name] = value
    return value

File: client/src/utils/parameterStore.js
SSM = require("@aws-sdk/client-ssm");
const client = new SSM.SSMClient({ region: "ap-southeast-2" });

const paramatersCache = {};

async function fetchParameterAws(parameter_name) {
    try {
        response = await client.send(
        new SSM.GetParameterCommand({
            Name: parameter_name
        }));
    return response.Parameter.Value;
    } catch (error) {
        return null;
    }
}

async function fetchParameterLocal(parameter_name) {
    if (parameter_name in paramatersCache) {
        return paramatersCache[parameter_name];
    } else {
        const parameterValue = await fetchParameterAws(parameter_name);
        if (parameterValue !== null) {
            paramatersCache[parameter_name] = parameterValue;
        }
        return parameterValue;
    }
}

export default fetchParameterLocal;